apps/codeimage/src/components/Presets/PresetSwitcher/PresetSwitcher.tsx

```diff
@@ -34,11 +35,55 @@ import {EmptyPresetFallback} from '../EmptyPresetFallback/EmptyPresetFallback';
 import {PresetPreview} from '../PresetPreview/PresetPreview';
 import {PresetUpdateDialog} from '../PresetUpdateDialog';
 import * as styles from './PresetSwitcher.css';
+import {getAuth0State} from '@codeimage/store/auth/auth0';
 
 type PresetSwitcherProps = {
   onClose: () => void;
 };
+const env = import.meta.env;
+const userLimit = env.VITE_PRESET_LIMIT;
+const guestLimit = env.VITE_PRESET_LIMIT_GUEST;
+function AddNewPreset() {
+  const presetsStore = getPresetsStore();
+  const openDialog = createControlledDialog();
+  const [t] = useI18n<AppLocaleEntries>();
+  const isLogged = () => getAuth0State().loggedIn();
+  const guestLimitReach = () =>
+    !isLogged() && presetsStore.sortedPresets().length >= guestLimit;
+  const userLimitReach = () =>
+    isLogged() && presetsStore.sortedPresets().length >= userLimit;
 
+  const isSomeLimitReach = () => guestLimitReach() || userLimitReach();
+  const labelTooltip = () =>
+    isLogged()
```

Comment:
We have to add this label into translation files. In my opinion we should use the same message for both cases

Something like:
```
Preset limits (<size>) reached
```

Reply:
in case the user is gues we should invite them to log in. Actually, those were placeholder labels, but you get the point I'd like to return two different labels for each case. should we not?

Reply:
Yes we can differentiate the labels